### PR TITLE
Add install-concept.md

### PR DESCRIPTION
**Pull request by joshmue**
First draft of documentation discussed in the last SIG meeting ([minutes](https://github.com/SovereignCloudStack/minutes/blob/main/sig-central-api/20231205.md)).

### CODE REVIEW

**Comment by joshmue** [comment]
@artificial-intelligence @scoopex @janiskemper

Please feel free to give feedback regarding expressiveness and the general direction.

File: install-concept.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+6. Make all required changes to the central-api cluster to serve as central-api (this is where most automation will come in to place credentials for the tenant's Openstack project, for the tenant's keycloak realm etc.)
+```mermaid
+flowchart LR
+    B(CSP CAPI mgmt cluster) -->|Manage| C(CSP central-api cluster)
+    B -->|Manage| B
+    C -->|Install central-api components, establish access, setup RBAC etc.| C
+```
+
+7. Actually make the central-api acessible to tenants, to manage their resources  (in tenant Openstack project using application credentials)
```
Comment (artificial-intelligence) [inline install-concept.md]:
```suggestion
7. Actually make the central-api accessible to tenants, to manage their resources  (in tenant Openstack project using application credentials)
```

**Review by artificial-intelligence — COMMENTED** [review]
In general I like these docs for a poc.
For a proper release I think we should also add some rational why things are done this way, design considerations etc.

Not sure if we want to add this stuff already now or if it's fine to change this later.

**Comment by scoopex** [comment]
I will review this on monday :-)

File: install-concept.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    A[CSP Bootstrap host] -->|install kind+CAPI| A
+```
+
+2. Use bootstrapping kind+CAPI to create Mgmt cluster (in e. g. CSP hidden Openstack project using application credentials)
+```mermaid
+flowchart LR
+    A[CSP Bootstrap host with kind+CAPI] -->|Create| B(CSP CAPI mgmt cluster)
+```
+
+3. `clusterctl move` to new cluster, which should start managing itself
```
Comment (janiskemper) [inline install-concept.md]: From CAPI perspective you create a bootstrap cluster with kind, then a workload cluster (i.e. a proper Kubernetes cluster) and then move the management components from the bootstrap cluster in there. I feel like the current state is a bit misleading in this regard ;)

File: install-concept.md [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+```
+
+4. Remove bootstrapping host
+```mermaid
+flowchart TB
+    A[CSP Bootstrap host with kind+CAPI] -->|Delete| A
+    B(CSP CAPI mgmt cluster) -->|Manage| B
+```
+
+5. Use CAPI mgmt cluster to create central-api cluster (in e. g. CSP hidden Openstack project using application credentials)
```
Comment (janiskemper) [inline install-concept.md:26]: Here we talked about the following two options:
1. use the CAPI management cluster for the central api as well
2. create a separate cluster for central API

You described the second option. It has the drawback, that parts of the central API are related to the Kubenretes management itself, right? And this is then in another cluster (the CAPI management cluster), separated from the IaaS management

**Review by janiskemper — COMMENTED** [review]
one improvement suggestion, another broader discussion that shouldn't be hidden!

File: install-concept.md (hunk 2 above)
Comment (joshmue) [reply install-concept.md]: Reworded this a bit and added multiple references to upstream docs. Do you think that improved the situation?

File: install-concept.md (hunk 3 above)
Comment (joshmue) [reply install-concept.md:26]: > Here we talked about the following two options:
> 
>     1. use the CAPI management cluster for the central api as well
>     2. create a separate cluster for central API
> 
> You described the second option.

Yes.

> It has the drawback, that parts of the central API are related to the Kubenretes management itself, right? And this is then in another cluster (the CAPI management cluster), separated from the IaaS management

I do not understand this one exactly. Could you elaborate?

Generally, CAPI components would be duplicated - but for completely different concerns: First, management of the central-api cluster (done completely by CSP's) and second, management of tenant clusters (used by tenants).

**Comment by joshmue** [comment]
@artificial-intelligence:

> For a proper release I think we should also add some rational why things are done this way, design considerations etc.

Definitely.

> Not sure if we want to add this stuff already now or if it's fine to change this later.

This PR was meant to solely create a foundation for a common understanding of the to-be-implemented installation concept, so I'd create a distinct PR for some rational.

File: install-concept.md (hunk 3 above)
Comment (janiskemper) [reply install-concept.md:26]: okay, I'm not sure what you mean by having "CAPI components duplicated". The management components cannot be duplicated AFAIK in the sense that they run in multiple clusters, i.e. having multiple CAPI management clusters.

I understand though that you intend on having the cluster API management resources in a cluster that is accessible for users and the central API resources (e.g. for IaaS) in another cluster. 

If we now consider the CAPI resources (and cluster stack resources) part of the wider central API scope, then we have two clusters with central API resources. That's what I meant ;) 

If that is a design that is intended, it's okay! I just wanted to point out that Kubernetes-related resources would be then separated from IaaS-related resources of the Central API.

Comment (joshmue) [reply install-concept.md:26]: > okay, I'm not sure what you mean by having "CAPI components duplicated". The management components cannot be duplicated AFAIK in the sense that they run in multiple clusters, i.e. having multiple CAPI management clusters.

Not for managing the same set of clusters, AFAIU, yes :+1: 

> I understand though that you intend on having the cluster API management resources in a cluster that is accessible for users and the central API resources (e.g. for IaaS) in another cluster.
> ...
> If that is a design that is intended, it's okay! I just wanted to point out that Kubernetes-related resources would be then separated from IaaS-related resources of the Central API.

Not really; What do you mean with separation between CAPI resources and IaaS resources?
As it stands now, CAPI is (1) an implementation detail on how to provision the central-api cluster without leaking too much into the "workload" that is the central-api cluster and (2) CRD's+controller on the central-api cluster that are usable by users alongside any (Crossplane) IaaS/IAM/... CRD's+controllers residing on the same central-api cluster.

> If we now consider the CAPI resources (and cluster stack resources) part of the wider central API scope, then we have two clusters with central API resources. That's what I meant ;)

Technically yes :D

**Comment by bitkeks** [comment]
LGTM for first draft to get some workflow graphics going.

@janiskemper @joshmue should we follow up on your conversation in https://github.com/SovereignCloudStack/central-api/pull/3#discussion_r1425366199?

**Review by bitkeks — APPROVED** [review]
LGTM for now, please continue the started conversation @janiskemper @joshmue

**Comment by joshmue** [comment]
Working on a PR with design considerations and reasoning.

**Pull request merged**